Library/OcVariableLib/VariableManagement.c

```diff
@@ -178,7 +220,7 @@ DeleteVariables (
       TmpBuffer = AllocateZeroPool (RequestedSize);
       if (TmpBuffer != NULL) {
         if (Buffer != NULL) {
-          CopyMem (TmpBuffer, Buffer, BufferSize);
+          CopyMem (TmpBuffer, Buffer, BufferSize); ///< TODO: Is this copy required?
```

Comment:
Why not?

Reply:
Why? You re-read the entire variable again into the new buffer no? (The copy makes it look as if it will just read the rest of the variable, beyond what was read before into the too-small buffer; but the call to read the variable isn't even passed enough info (i.e. the old size as well as the new size) for this to be possible, I think.)

Reply:
Doh! Sorry.